azurerm/internal/services/web/resource_arm_app_service_hybrid_connection.go

```diff
@@ -0,0 +1,205 @@
+		Read:   resourceArmAppServiceHybridConnectionRead,
+		Update: resourceArmAppServiceHybridConnectionCreateUpdate,
+		Delete: resourceArmAppServiceHybridConnectionDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"app_service_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAppServiceName,
+			},
+			"resource_group_name": azure.SchemaResourceGroupName(),
+			"relay_arm_uri": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+			"namespace_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"relay_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hostname": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"port": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validate.PortNumberOrZero,
+			},
+			"service_bus_namespace": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[a-zA-Z][-a-zA-Z0-9]{0,100}[a-zA-Z0-9]$"),
+					"The namespace can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number.",
+				),
+			},
+			"service_bus_suffix": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"send_key_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"send_key_value": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+		},
+	}
+}
+
+func resourceArmAppServiceHybridConnectionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("app_service_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	relayArmURI := d.Get("relay_arm_uri").(string)
+	relayId, err := azure.ParseAzureResourceID(relayArmURI)
+	if err != nil {
+		return fmt.Errorf("Error parsing relay_arm_uri as ID: %s", relayArmURI)
+	}
+
+	namespaceName := relayId.Path["namespaces"]
+	if namespaceName == "" {
+		return fmt.Errorf("Failed to parse relay namespace from ID: %s", relayArmURI)
+	}
+
+	relayName := relayId.Path["hybridConnections"]
+	if relayName == "" {
+		return fmt.Errorf("Failed to parse relay name from ID: %s", relayArmURI)
+	}
```

Comment:
So since you added the checks you might as add a parse ID function for that resource! We are planning on slowly creating an ID and parse function for each resource ID so we get these checks & import id validation. Here is an example in the code right now:

```
type AppServiceResourceID struct {
	ResourceGroup string
	Name          string
}

func ParseAppServiceID(input string) (*AppServiceResourceID, error) {
	id, err := azure.ParseAzureResourceID(input)
	if err != nil {
		return nil, fmt.Errorf("[ERROR] Unable to parse App Service ID %q: %+v", input, err)
	}

	appService := AppServiceResourceID{
		ResourceGroup: id.ResourceGroup,
	}

	if appService.Name, err = id.PopSegment("sites"); err != nil {
		return nil, err
	}

	if err := id.ValidateNoEmptySegments(input); err != nil {
		return nil, err
	}

	return &appService, nil
}
```

this will also simplify things elsewhere